# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/MediaEngineTabVideoSource.cpp
# Commit: 20543d03a4cd
# Full Hash: 20543d03a4cda05c785011a25079aa42aec8b2ba
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2019-03-29 21:53:01
# Regressor Bug: 1494675
# File Overlap Count: 2
# Description:
#   Bug 1494675 - Remove AllocationHandle. r=padenot
#   
#   The handle was used to keep separate allocations of the same source in a single
#   process apart. Sources no longer use sharing, so we no longer need allocations
#   or their handles even as a concept.
# ==============================================================================

diff -r 9e7832fee028 -r 20543d03a4cd dom/media/webrtc/MediaEngineTabVideoSource.cpp
--- a/dom/media/webrtc/MediaEngineTabVideoSource.cpp	Wed Mar 27 14:10:27 2019 +0000
+++ b/dom/media/webrtc/MediaEngineTabVideoSource.cpp	Wed Mar 27 14:10:24 2019 +0000
@@ -18,7 +18,6 @@
 #include "nsPresContext.h"
 #include "gfxContext.h"
 #include "gfx2DGlue.h"
-#include "AllocationHandle.h"
 #include "ImageContainer.h"
 #include "Layers.h"
 #include "nsIInterfaceRequestorUtils.h"
@@ -138,7 +137,7 @@
     const dom::MediaTrackConstraints& aConstraints,
     const MediaEnginePrefs& aPrefs, const nsString& aDeviceId,
     const mozilla::ipc::PrincipalInfo& aPrincipalInfo,
-    AllocationHandle** aOutHandle, const char** aOutBadConstraint) {
+    const char** aOutBadConstraint) {
   AssertIsOnOwningThread();
 
   // windowId is not a proper constraint, so just read it.
@@ -147,20 +146,16 @@
   mWindowId = aConstraints.mBrowserWindow.WasPassed()
                   ? aConstraints.mBrowserWindow.Value()
                   : -1;
-  *aOutHandle = nullptr;
   mState = kAllocated;
 
-  return Reconfigure(nullptr, aConstraints, aPrefs, aDeviceId,
-                     aOutBadConstraint);
+  return Reconfigure(aConstraints, aPrefs, aDeviceId, aOutBadConstraint);
 }
 
 nsresult MediaEngineTabVideoSource::Reconfigure(
-    const RefPtr<AllocationHandle>& aHandle,
     const dom::MediaTrackConstraints& aConstraints,
     const mozilla::MediaEnginePrefs& aPrefs, const nsString& aDeviceId,
     const char** aOutBadConstraint) {
   AssertIsOnOwningThread();
-  MOZ_ASSERT(!aHandle);
   MOZ_ASSERT(mState != kReleased);
 
   // scrollWithPage is not proper a constraint, so just read it.
@@ -214,10 +209,8 @@
   return NS_OK;
 }
 
-nsresult MediaEngineTabVideoSource::Deallocate(
-    const RefPtr<const AllocationHandle>& aHandle) {
+nsresult MediaEngineTabVideoSource::Deallocate() {
   AssertIsOnOwningThread();
-  MOZ_ASSERT(!aHandle);
   MOZ_ASSERT(mState == kAllocated || mState == kStopped);
 
   if (mStream && IsTrackIDExplicit(mTrackID)) {
@@ -231,7 +224,6 @@
 }
 
 void MediaEngineTabVideoSource::SetTrack(
-    const RefPtr<const AllocationHandle>& aHandle,
     const RefPtr<SourceMediaStream>& aStream, TrackID aTrackID,
     const mozilla::PrincipalHandle& aPrincipal) {
   AssertIsOnOwningThread();
@@ -247,8 +239,7 @@
   mStream->AddTrack(mTrackID, new VideoSegment());
 }
 
-nsresult MediaEngineTabVideoSource::Start(
-    const RefPtr<const AllocationHandle>& aHandle) {
+nsresult MediaEngineTabVideoSource::Start() {
   AssertIsOnOwningThread();
   MOZ_ASSERT(mState == kAllocated);
 
@@ -386,13 +377,11 @@
   mStreamMain->AppendToTrack(mTrackIDMain, &segment);
 }
 
-nsresult MediaEngineTabVideoSource::FocusOnSelectedSource(
-    const RefPtr<const AllocationHandle>& aHandle) {
+nsresult MediaEngineTabVideoSource::FocusOnSelectedSource() {
   return NS_ERROR_NOT_IMPLEMENTED;
 }
 
-nsresult MediaEngineTabVideoSource::Stop(
-    const RefPtr<const AllocationHandle>& aHandle) {
+nsresult MediaEngineTabVideoSource::Stop() {
   AssertIsOnOwningThread();
 
   if (mState == kStopped || mState == kAllocated) {